src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -85,8 +85,23 @@ The storage type of ``relu`` output depends upon the input storage type:
 )code" ADD_FILELINE)
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseOut{"_backward_relu"});
 
-MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU(_backward_relu,
-                                               unary_bwd<mshadow_op::relu_grad>);
+MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU(_backward_relu, unary_bwd<mshadow_op::relu_grad>)
+.set_attr<nnvm::FGradient>("FGradient",
+    [](const nnvm::NodePtr& n, const std::vector<nnvm::NodeEntry>& ograds) {
+      std::vector<nnvm::NodeEntry> ret;
+      // ograds[0]: d^2L/dx^2
+      // inputs[0]: dL/dy
+      // inputs[1]: y
+      // f(x) -> relu(x)
+      // f'(x) = 1 if x > 0 else 0
+      // f''(x) = 0
+      auto gx = nnvm::NodeEntry{n};  // f'(x)
+      ret.emplace_back(MakeNode("elemwise_mul", n->attrs.name + "_backward_grad_grad",
+                                {ograds[0], gx}, nullptr, &n));
```

Comment:
`gx` is actually `f'(x) * head_grads/output_gradient`.  
It should actually only be `f'(x)`.

Explanation :  gx = `f'(x) * head_grads`.
Therefore, `gx w.r.t. f'(x) = head_grads`,
Similarly `gx w.r.t. head_grads = f'(x)`.

Reply:
Updated